agent/agent.go

```diff
@@ -1004,6 +999,13 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 		}
 	}
 
+	// Attempt automatic task output ingestion if the task output directory
+	// was setup, regardless of the task status.
+	if tc.taskConfig != nil && tc.taskConfig.TaskOutputDir != nil {
+		tc.logger.Execution().Error(errors.Wrap(a.uploadTraces(ctx, tc.taskConfig.WorkDir), "uploading traces"))
+		tc.logger.Execution().Error(errors.Wrap(tc.taskConfig.TaskOutputDir.Run(ctx), "ingesting task output"))
+	}
```

Comment:
I didn't realize the upload sometimes takes so long.
Maybe we want to add a line to the task and/or agent log(s) saying we're uploading task logs so people don't wonder why their task isn't completing?

Reply:
This is for test logs (not task) logs. But we do log that! See [this task](https://spruce.mongodb.com/task/mongodb_mongo_master_enterprise_amazon_linux2_arm64_all_feature_flags_aggregation_aa282db5cd0b6d158204b5fffe681c00afca5480_24_05_10_02_33_39/logs?execution=0&sortBy=STATUS&sortDir=ASC) for example.

Reply:
I think for larger logs its on the order of a few minutes (we do parallelize the uploading). But in most case it is less than a minute.